src/include/distributed/metadata_cache.h

```diff
@@ -38,11 +38,12 @@ typedef struct
 
 	/* pg_dist_shard metadata (variable-length ShardInterval array) for this table */
 	int shardIntervalArrayLength;
-	ShardInterval *shardIntervalArray;
+	ShardInterval **sortedShardIntervalArray;
```

Comment:
I was wondering about this actually. Why did you change this? To make the sorting cheaper?   There's a disadvantage: It makes binary search slower. Direct value lookups are less expensive than lookups via an indirection (pointer).